src/main/java/com/google/sps/data/BusinessesService.java

```diff
@@ -111,4 +128,109 @@ private void addListingToBusinesses(PlacesSearchResult place) {
+    } catch(InterruptedException e) {
+      LOGGER.warning(e.getMessage());
+    } 
+  }
+  
+  private void checkBusinessThroughLinkedin(String currentBusinessName, 
+                          PlacesSearchResult[] similarBusinessesInTheArea) 
+                              throws GeneralSecurityException, IOException {
+    String cx = "REDACTED"; 
+    Customsearch cs = new Customsearch.Builder(
+                                GoogleNetHttpTransport.newTrustedTransport(), 
+                                JacksonFactory.getDefaultInstance(), 
+                                null) 
+                                .setApplicationName("linkedinSearch") 
+                                .setGoogleClientRequestInitializer(
+                                      new CustomsearchRequestInitializer(KEY)) 
+                                .build();
+
+    Customsearch.Cse.List list = cs.cse().list(currentBusinessName).setCx(cx); 
+    List<Result> searchJsonResults = list.execute().getItems();
+    String[] numberOfFollowers;
+    int companyFollowers = 0;
+    if (searchJsonResults!=null && searchJsonResults.size() != 0) {
+      Result linkedinBusiness = searchJsonResults.get(0);
+      String businessDescription = (String) linkedinBusiness.getPagemap()
+                                                            .get("metatags")
+                                                            .get(0)
+                                                            .get("og:description");
+      if(businessDescription.indexOf(startIndex) != -1 
+          && businessDescription.indexOf(endIndex) != -1){
+        String followers = businessDescription.substring(
+                                businessDescription.indexOf(startIndex) + 2, 
+                                businessDescription.indexOf(endIndex) - 1);
+        companyFollowers = 
+                    Integer.parseInt(followers.replaceAll(",", ""));
+      }
+      if (companyFollowers > minFollowers) {
+        addBigBusinessToDatabase();
+      } else {
+        checkNumberOfSimilarBusinessesInTheArea(currentBusiness.getName(), similarBusinessesInTheArea);
+      }
+    }
+  }
+
+  private void checkNumberOfSimilarBusinessesInTheArea(String businessName, 
+                          PlacesSearchResult[] similarBusinessesInTheArea) {
+    int countNumberOfMatchingBusiness = 0;
+    int i = 0;
+    while (i < similarBusinessesInTheArea.length 
+          && countNumberOfMatchingBusiness < 10) {
+      if(similarBusinessesInTheArea[i].vicinity != null){
+        if (similarBusinessesInTheArea[i].name.contains(businessName) 
+            && !(similarBusinessesInTheArea[i].vicinity.equals(currentBusiness.getFormattedAddress()))) {
+          countNumberOfMatchingBusiness++;
+        }
+      }
+      i++;
+     }
+     if (countNumberOfMatchingBusiness >= 10) {
+       addBigBusinessToDatabase();
+     }
+   }
+
+  private void addBigBusinessToDatabase(){
+    businesses.remove();
```

Comment:
Confused. What's being removed here? The entire list of businesses?

Reply:
This removes the current business that is determined to be a a big business, if you follow the code from the while loop in the checkNumberOfLocationsOfBusiness method. I think you would conclude that the same thing. I can add a comment, or  create a method that makes it more clear. I am doing this, so I do not need to use the removeBigBusinessesFromResults method again, to connect to the database and waste more time.

Reply:
This seems misplaced. The method is should focus on adding entries to the database. Please move it to the point where this method is called. I've also noted that this variable can be removed. Please see my previous comments.

Reply:
Yes I fixed this issue